### PR TITLE
Deleting data and files while archiving and deleting projects

### PR DESCRIPTION
**Pull request by aanil**
* Does not touch the ProjectStatuses table - Should we keep the status history or only keep the latest status(which would be deleted or archived).
* Tests updated.

### CODE REVIEW

**Comment by aanil** [comment]
Added the special case to be able to abort a project if it has been made available.

File: dds_web/api/project.py [hunk 1]
```diff
@@ -115,38 +116,56 @@ def post(self):
...
         )
+        delete_message = ""
         try:
             project.project_statuses.append(add_status)
-            db.session.commit()
-            if new_status in ["Deleted", "Archived"]:
-                # TODO Call function to delete files
-                print("TODO", flush=True)
+            if not project.is_active:
+                # Deletes files (also commits session in the function - should it be decoupled?)
```
Comment (inaod568) [inline dds_web/api/project.py:142]: We can keep it like it is for now and possibly refactor later 👍🏻

File: dds_web/api/project.py [hunk 2]
```diff
@@ -284,43 +323,35 @@ def delete(self):
...
-            if not removed:
-                raise DeletionError(
-                    message="No project contents deleted.",
-                    username=current_user.username,
-                    project=project.public_id,
-                )
+        if not project.files:
+            raise EmptyProjectException("The are no project contents to delete.")
+        try:
+            delete_project_contents(project)
```
Comment (inaod568) [inline dds_web/api/project.py:329]: `self.delete_project_contents(project)`

**Review by inaod568 — CHANGES_REQUESTED** [review]
Looks good! Just a couple of small changes

File: dds_web/api/project.py (hunk 1 above)
Comment (aanil) [reply dds_web/api/project.py]: 👍

File: dds_web/api/project.py (hunk 2 above)
Comment (aanil) [reply dds_web/api/project.py]: Fixed!

**Review by inaod568 — APPROVED** [review]

**Comment by inaod568** [comment]
> * Does not touch the ProjectStatuses table - Should we keep the status history or only keep the latest status(which would be deleted or archived).
> 
>     * Tests updated.

Can you update this description a little bit? Thanks!

**Pull request merged**